pj-dao/src/contract.rs

```diff
@@ -99,7 +98,8 @@ pub fn admin_mint<S: Storage, A: Api, Q: Querier>(
 pub fn join_dao<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
     env: Env,
-    nft: Option<JoinNftDetails>,
+    nft_id: Option<String>,
+    permit: Permit,
```

Comment:
This needs to be an option, just like the nft_id, we do not use it if nft_id is not there. Otherwise will cause frontend changes.